lib/crowbar/init/application.rb

```diff
@@ -471,13 +493,17 @@ class Application < Sinatra::Base
 
           if init[:body] # nil body means success
             logger.error(init[:body])
-            res[:error] = init[:body]
+            res[:errors] = {
+                data: init[:body]
+            }
             http_code = 422
           end
         end
 
-        if result[:error]
-          upgrade_status.end_step(false, database: result[:error])
+        if result[:errors] || http_code != 200
+          # reformat the error response to match common format
+          result[:errors] = { database_connect: result[:errors] }
```

Comment:
If I modify the format of errors to be embedded inside the database "steps" then probably checking for errors will be too cumbersome so I'll only leave `http_code`.